presto-verifier/src/main/java/com/facebook/presto/verifier/resolver/FailureResolverUtil.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.resolver;
+
+import com.facebook.presto.spi.ErrorCodeSupplier;
+import com.facebook.presto.verifier.framework.PrestoQueryException;
+import com.facebook.presto.verifier.framework.QueryException;
+import com.facebook.presto.verifier.framework.QueryStage;
+
+import java.util.Optional;
+
+public class FailureResolverUtil
+{
+    private FailureResolverUtil() {}
+
+    public static boolean isMatchingPrestoException(QueryException queryException, QueryStage queryStage, ErrorCodeSupplier errorCode)
```

Comment:
@caithagoras Based on the usage of this method, I think a signature with a callback may work better. 

```
    public static <T> Optional<T> mapMatchingPrestoException(
            QueryException queryException,
            QueryStage queryStage,
            ErrorCodeSupplier errorCode,
            Function<PrestoQueryException, T> mapper)
    {
        if (queryException.getQueryStage() != queryStage || !queryException.isPrestoQueryException()) {
            return Optional.empty();
        }

        return Optional.of(mapper.apply((PrestoQueryException) queryException));
    }

public class ExceededGlobalMemoryLimitFailureResolver
        implements FailureResolver
{
    public static final String NAME = "exceeded-global-memory-limit";

    @Override
    public Optional<String> resolve(QueryStats controlQueryStats, QueryException queryException, Optional<QueryBundle> test)
    {
        return mapMatchingPrestoException(queryException, TEST_MAIN, EXCEEDED_GLOBAL_MEMORY_LIMIT,
                e -> "Control query uses more memory than the test cluster memory limit");
    }
}

```

Reply:
With `isMatchingPrestoException`, I meant to do an initial check. With the callback method however, the mapper function needs to contain all the the logic necessary to determine whether to resolve in additional to the initial check. In case of `TooManyOpenPartitionsFailureResolve`, this is going to be much more complex.

Reply:
Comments addessed.